### PR TITLE
#21579 delete audit when deleting all bundle elements from queue

### PR DESCRIPTION
**Pull request by dsilvam**
This deletes the bundle from the audit table if all its elements were removed from the publishing_queue. This is to avoid infinite retries.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java [hunk 1]
```diff
@@ -394,6 +397,13 @@ public void deleteAssetFromBundle(String assetId, String bundleId)
 			throws DotDataException {
 		bundleFactory.deleteAssetFromBundle(assetId, bundleId);
 
+		List<PublishQueueElement> queueElements = Try.of(()->PublisherAPI.getInstance()
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java:400]: set to final

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java [hunk 2]
```diff
@@ -731,6 +733,10 @@ public void deleteElementFromPublishQueueTable(final String identifier) throws D
 	@Override
 	public void deleteElementFromPublishQueueTable(final String identifier, final long languageId) throws DotPublisherException{
 
+		final List<PublishQueueElement> queueElements = getQueueElementsByAsset(identifier);
+
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:737]: set to final

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java [hunk 3]
```diff
@@ -743,10 +749,16 @@ public void deleteElementFromPublishQueueTable(final String identifier, final lo
 			}
 			dc.loadResult();
 		}catch(Exception e){
-
 			Logger.error(PublisherUtil.class,e.getMessage(),e);
 			throw new DotPublisherException("Unable to delete element "+identifier+" :"+e.getMessage(), e);
 		}
+
+		List<PublishQueueElement> queueElementsAfterDelete = Try.of(()->PublisherAPI.getInstance()
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:756]: the PublisherAPI is not being retrieved by APILocator, if not it shouldn't?

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — CHANGES_REQUESTED** [review]
This is breaking some tests: https://raw.githack.com/dotCMS/test-results/issue-21579-retry-the-configured-times/projects/core/current/curl/postgres/reports/html/Bundle_Resource.html

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:756]: It never does. It's referenced everywhere like this

**Pull request closed without merging**